youtube_dl/extractor/kaltura.py

```diff
@@ -139,6 +140,13 @@ def _extract_url(webpage):
                 url = smuggle_url(url, {'service_url': service_url.group(1)})
             return url
 
+        # Check for an iframe, which may require redirection.
+        mobj = re.search(
+            r"<iframe[^>]+src=['\"](?P<url>(https?:)?//www\.kaltura\.com/[^'\"]+)['\"]",
```

Comment:
OK.

It seemed more robust to follow the redirect and use the existing parser. For instance, this test, `http://www.gsd.harvard.edu/event/i-m-pei-a-centennial-celebration/` contains (linebreaks in URLs by me):

```
<iframe
  src="//www.kaltura.com/p/1487951/sp/148795100/embedIframeJs/uiconf_id/26739141
  /partner_id/1487951?iframeembed=true&playerId=kaltura-video-embed-&
  entry_id=0_f2cfbpwy"
  width="1280" height="720" allowfullscreen webkitallowfullscreen
  mozAllowFullScreen frameborder="0">
```

This redirects to:

```
Location: http://cdnapi.kaltura.com/html5/html5lib/v2.46/mwEmbedFrame.php/
  p/1487951/uiconf_id/26739141/entry_id/0_f2cfbpwy?wid=_1487951&
  iframeembed=true&playerId=kaltura-video-embed-1&entry_id=0_f2cfbpwy
```

But I can certainly pull out the `&entry_id=0_f2cfbpwy` from the iframe src.
It just seemed more fragile.